### PR TITLE
improve gen-libs for Android platform

### PR DESCRIPTION
**Pull request by drelaptop**
fix, https://github.com/cocos2d/cocos2d-x/issues/18790

### CODE REVIEW

File: plugins/plugin_generate/gen_libs.py [hunk 1]
```diff
@@ -346,26 +347,27 @@ def compile_android(self):
...
-        obj_dir = os.path.join(proj_path, ANDROID_A_PATH)
-        copy_cfg = {
-            "from": obj_dir,
-            "to": android_out_dir,
-            "include": [
-                "*.a$"
-            ]
-        }
-        cocos.copy_files_with_config(copy_cfg, obj_dir, android_out_dir)
+        for app_abi_item in self.app_abi_list:
```
Comment (drelaptop) [inline plugins/plugin_generate/gen_libs.py:350]: Fix error when set multi ABI, for example:
```
cocos gen-libs -p android --app-abi armeabi-v7a:x86
```

File: bin/strings.json [hunk 2]
```diff
@@ -141,6 +141,7 @@
         "COMPILE_ARG_TOOLCHAIN" : "Specify the NDK_TOOLCHAIN of ndk-build.",
         "COMPILE_ARG_CPPFLAGS" : "Specify the APP_CPPFLAGS of ndk-build.",
         "COMPILE_ARG_NO_APK" : "Not generate the apk file.",
+        "COMPILE_ARG_NO_SIGN" : "Generate unsigned apk even when release mode.",
```
Comment (drelaptop) [inline bin/strings.json:144]: Support to generate unsigned apk, it's meanful when `gen-libs`. Another benefit, developer can test release compile with no sign input.

File: plugins/plugin_compile/build_android.py [hunk 3]
```diff
@@ -551,22 +551,27 @@ def do_build_apk(self, mode, no_apk, output_dir, custom_step_args, android_platf
...
 
             # build apk
             self.gradle_build_apk(mode, android_platform, compile_obj)
 
             # copy the apk to output dir
             if output_dir:
-                apk_name = '%s-%s.apk' % (project_name, mode)
+                # support generate unsigned apk
+                if mode == "release" and no_sign:
+                    apk_name = '%s-%s-unsigned.apk' % (project_name, mode)
```
Comment (drelaptop) [inline plugins/plugin_compile/build_android.py:565]: unsigned apk have its special suffix, for example
```
CppEmptyTest-release-unsigned.apk
```

**Comment by drelaptop** [comment]
please review and merge @minggo 

correspond issue, https://github.com/cocos2d/cocos2d-x/issues/18790

**Comment by minggo** [comment]
key is only needed for Android APK, it is not needed for generating libs, so i don't think the parameter is needed.

**Comment by drelaptop** [comment]
Yes, it's not needed for gen-libs, but now `cocos gen-libs -p android -m release` need user to input key info. so add a `no-sign` to avoid `gen-libs` to use keystore file.

**Comment by minggo** [comment]
what i mean is that, should fix the logic to avoid checking key info instead of adding a new parameter.

**Comment by drelaptop** [comment]
the `cocos gen-libs` is finished by call `cocos compile` cmd, and copy the libs to some dir, add a new parameter to `cocos compile` is a convenient way for `cocos gen-libs` to control. And this new parameter didn't break the older compile logic. 

find another way to fix the logic will cause to more code changes, I will find another way to compare with this way if you stick to.

**Comment by minggo** [comment]
Yep, i think should put the logic of compiling libs in one function, for example named `compile_cpp_codes()`, then `cocos gen-libs` and `cocos compile` can use the same logic.

**Comment by drelaptop** [comment]
after I review the `project_compile.py` and `build_android.py` again, I think it's not a good choice to put the compile logic into a function. 

the implement now use many member variables, which values is gotten from `cocos compile` cmd line. to put all compile logic into a function, and then call when gen libs, so that we will have to set members values manually in `gen_libs.py`. Another reason I think `--no-sign` is a good optional parameter for android compile, sign infos are not necessary for Android release compile process, it's only necessary when run release apk into android device. __Gradle don't ask developer to input sign infos before release compile process, but cocos console asked.__

**Comment by crazyhappygame** [comment]
@minggo PR looks good for me. Could you merge it?

**Pull request merged**